almdrlib/client.py

```diff
@@ -619,7 +694,52 @@ def __call__(self, *args, **kwargs):
         return self._call(*args, **kwargs)
 
     def __repr__(self):
-        return f"<{type(self).__name__}: [{self._method}] {self._path}>"
+        return f"<{self._client.name}.{self.operation_id}: " \
+               f"{self._method.upper()} {self._path}>"
+
+    @property
+    def __signature__(self):
+        if self._signature is None:
+            required_params = []
+            body_params = []
+            non_required_params = []
+            # Define the path, query, header, and cookie parameters
+            for p in self._params:
+                if p.required:
+                    required_params.append(p.to_inspect_parameter())
+                else:
+                    non_required_params.append(p.to_inspect_parameter())
+            # Define the body parameter, if it exists
+            if self.body is not None:
+                # Define the content type parameter if there's more than one
+                if not self.body.default_content_type:
+                    param = inspect.Parameter(
+                        OpenAPIKeyWord.CONTENT_TYPE_PYTHON_PARAM,
+                        inspect.Parameter.KEYWORD_ONLY)
+                    body_params.append(param)
+                # Define each possible body parameter.  Note that these may be
+                # mutually exclusive.
+                for body_param in self.body.parameters.keys():
+                    param = inspect.Parameter(body_param,
+                                              inspect.Parameter.KEYWORD_ONLY)
+                    body_params.append(param)
```

Comment:
This part is a bit unclean.  The body can be specified by potentially multiple different variables, some of which may be marked as required.  This just treats any body field name as optional (since that's the OpenAPI default) and I couldn't find any alsdkdef that had a required body and adds all possible body field names.

This also adds the `content_type` param if necessary.